crates/acvm/src/lib.rs

```diff
@@ -60,7 +77,9 @@ pub trait PartialWitnessGenerator {
                     // We compute the result because the other gates may want to use the assignment to generate their assignments
                 }
                 Gate::GadgetCall(gc) => {
-                    Self::solve_gadget_call(initial_witness, gc)?;
+                    if let Err(op) = Self::solve_gadget_call(initial_witness, gc) {
+                        return Err(GateResolution::UnsupportedOpcode(op));
```

Comment:
Looks like solve_gadget_call should return this directly instead of requiring a conversion

Reply:
I agree but that would require to modify the aztec-backend, and we already have several versions of it (wasm, native, with cli) so it's a bit messy, I prefer not to add to the confusion.
I can create an issue for this and I will resolve it after the next Noir release, is it OK?